packages/devtools_app/lib/src/screens/inspector_v2/inspector_controller.dart

```diff
@@ -818,6 +808,58 @@ class InspectorController extends DisposableController
     animateTo(selectedNode.value);
   }
 
+  static const _implementationWidgetMessage =
+      'Selected an implementation widget';
+
+  static const _notificationDuration = Duration(seconds: 4);
+
+  Future<void> _maybeShowNotificationForSelectedNode({
+    required RemoteDiagnosticsNode? selectedNode,
+    required ObjectGroup group,
+  }) async {
+    if (selectedNode == null) return;
+    if (!implementationWidgetsHidden.value) return;
+
+    // Return early if we have a new selected node.
+    if (_selectionIsOutOfDate(selectedNode)) return;
+
+    final possibleImplementationWidget = await group.getSelection(
+      selectedDiagnostic,
+      treeType,
+    );
+
+    // Return early if we have a new selected node.
+    if (_selectionIsOutOfDate(selectedNode)) return;
+
+    final isImplementationWidget =
+        possibleImplementationWidget != null &&
+        !possibleImplementationWidget.isCreatedByLocalProject;
+    if (isImplementationWidget) {
+      final selectedWidgetName = selectedNode.description ?? '';
+      final implementationWidgetName =
+          possibleImplementationWidget.description ?? '';
+
+      // Return early if we have a new selected node.
+      if (_selectionIsOutOfDate(selectedNode)) return;
+
+      final messageDetails =
+          selectedWidgetName.isEmpty
+              ? ''
+              : ' of $selectedWidgetName${implementationWidgetName.isEmpty ? '' : ': $implementationWidgetName'}';
+      notificationService.pushNotification(
+        NotificationMessage(
+          '$_implementationWidgetMessage$messageDetails.',
```

Comment:
can you add a comment describing an example what this message will end up looking like?